source/Mlos.Python/mlos/Spaces/HypergridAdapters/unit_tests/TestContinuousToPolynomialBasisHypergridAdapter.py

```diff
@@ -0,0 +1,210 @@
+    # dataframe projection tests
+    def test_balanced_hierarchical_hypergrid_degree_two_all_terms_dataframe(self):
+        adapter_kwargs = {'degree': 2, 'interaction_only': False}
+        self._test_dataframe_projection(adaptee=self.balanced_hierarchical_hypergrid, adapter_kwargs=adapter_kwargs, num_random_points=10)
+
+    def test_balanced_hierarchical_hypergrid_degree_three_all_terms_dataframe(self):
+        adapter_kwargs = {'degree': 3, 'interaction_only': False}
+        self._test_dataframe_projection(adaptee=self.balanced_hierarchical_hypergrid, adapter_kwargs=adapter_kwargs, num_random_points=10)
+
+    def test_balanced_hierarchical_hypergrid_degree_two_interactions_only_dataframe(self):
+        adapter_kwargs = {'degree': 2, 'interaction_only': False}
+        self._test_dataframe_projection(adaptee=self.balanced_hierarchical_hypergrid, adapter_kwargs=adapter_kwargs, num_random_points=10)
+
+    def test_balanced_hierarchical_hypergrid_degree_three_interactions_only_dataframe(self):
+        adapter_kwargs = {'degree': 3, 'interaction_only': False}
+        self._test_dataframe_projection(adaptee=self.balanced_hierarchical_hypergrid, adapter_kwargs=adapter_kwargs, num_random_points=10)
+
+    # point projection tests
+    def test_balanced_hierarchical_hypergrid_degree_two_all_terms_point(self):
+        adapter_kwargs = {'degree': 2, 'interaction_only': False}
+        self._test_point_projection(adaptee=self.balanced_hierarchical_hypergrid, adapter_kwargs=adapter_kwargs, num_random_points=10)
+
+    def test_balanced_hierarchical_hypergrid_degree_three_all_terms_point(self):
+        adapter_kwargs = {'degree': 3, 'interaction_only': False}
+        self._test_point_projection(adaptee=self.balanced_hierarchical_hypergrid, adapter_kwargs=adapter_kwargs, num_random_points=10)
+
+    def test_balanced_hierarchical_hypergrid_degree_two_interactions_only_point(self):
+        adapter_kwargs = {'degree': 2, 'interaction_only': False}
+        self._test_point_projection(adaptee=self.balanced_hierarchical_hypergrid, adapter_kwargs=adapter_kwargs, num_random_points=10)
+
+    def test_balanced_hierarchical_hypergrid_degree_three_interactions_only_point(self):
+        adapter_kwargs = {'degree': 3, 'interaction_only': False}
+        self._test_point_projection(adaptee=self.balanced_hierarchical_hypergrid, adapter_kwargs=adapter_kwargs, num_random_points=10)
+
+    def _test_dataframe_projection(self, adaptee, adapter_kwargs, num_random_points):
+        num_adaptee_continuous_dims = 0
+        for adaptee_dim in adaptee.dimensions:
+            if isinstance(adaptee_dim, ContinuousDimension):
+                num_adaptee_continuous_dims += 1
+
+        # count the number of polynomial terms expected excluding the constant term
+        num_target_continuous_dims_expected = self.n_choose_k(adapter_kwargs['degree'] + num_adaptee_continuous_dims, num_adaptee_continuous_dims) - 1
+        if adapter_kwargs['interaction_only']:

```

Comment:
if `adapter_kwargs['interaction_only']` is true, we are overwriting the results of computation in 169. Can we use an if-else?